src/main/java/core/option/UserColorsPanel.java

```diff
@@ -0,0 +1,340 @@
+							}
+							updateRow(tableSelection, hoColor);
+							OptionManager.instance().setOptionsChanged(true);
+						}
+					}
+				}
+			}
+		});
+		return box;
+	}
+
+	/**
+	 * Update the current row of the color settings table
+	 * @param color Color setting
+	 */
+	public void updateRow(HOColor color) {
+		updateRow(colorTable.getEditingRow(), color);
+		OptionManager.instance().setOptionsChanged(true);
+	}
+
+	/**
+	 * Update color settings row
+	 * @param row Row number
+	 * @param color Color setting
+	 */
+	private void updateRow(int row, HOColor color) {
+		var tableModel = (DefaultTableModel) colorTable.getModel();
+		tableModel.setValueAt(createNameLabel(color.getHOColorName()), row, 0);
+		tableModel.setValueAt(color.colorReference(), row, 1);
+		tableModel.setValueAt(color, row, 2);
+		tableModel.setValueAt(color.getDefaultValue(), row, 3);
+	}
+
+	/**
+	 * Reset color setting to given default setting
+	 * @param color Default color setting
+	 */
+	public void resetRow(HOColor color) {
+		colors.set(colorTable.getEditingRow(), color);
+		updateRow(color);
+	}
+
+	/**
+	 * Create label displaying the color name (1st column of the settings table)
+	 * The provided technical color name is translated to the select language.
+	 * This value is used as tool tip too.
+	 * @param colorName Color name
+	 * @return JLabel
+	 */
+	private JLabel createNameLabel(HOColorName colorName) {
+		String text = colorName!=null?colorName.toString():"";
+		var label = new JLabel(text);
+		label.setToolTipText(text);
+		return label;
+	}
+
+	/**
+	 * Get selected theme name
+	 * @return Theme name
+	 */
+	public String getSelectedTheme(){
+		return UserParameter.temp().skin;
+	}
+
+	/**
+	 * action
+	 */
+	public void actionPerformed(ActionEvent arg0) {
+		if (arg0.getSource() == skins) {
+			UserParameter.temp().skin = (String) skins.getSelectedItem();
+			initData(UserParameter.instance().skin);
+			// TODO change the look and feel dynamically
+			OptionManager.instance().setRestartNeeded();
+		}
+
+	}
+
+	/**
+	 * Store the edited color settings
+	 */
+	public void storeChangedColorSettings() {
+		for (var color : colors) {
+			var theme = getSelectedTheme();
+			var origValue = HOColor.getHOColor(color.getHOColorName(), theme);
+			assert origValue != null;
+			if (AreDifferentColors(origValue, color) ||
+					AreDifferentColors(origValue.getDefaultValue(), color.getDefaultValue())) {
+				if (!color.getTheme().equals("default")) {
+					color.setTheme(theme);
+					DBManager.instance().storeHOColor(color);
+				}
+				else {
+					// reset default
+					DBManager.instance().deleteHOColor(origValue);
+				}
+				HOColor.addColor(color);
+				OptionManager.instance().setRestartNeeded();
+			}
+		}
+	}
+
+	/**
+	 * Compare two color settings
+	 * @param c1 Color setting, may be null
+	 * @param c2 Color setting, may  be null
+	 * @return True, if the settings are different
+	 */
+	private boolean AreDifferentColors(HOColor c1, HOColor c2) {
```

Comment:
Can you please lowercase this method and the next one, and push into HOColor?

Reply:
done